apiserver/pkg/modelservice/modelservice.go

```diff
@@ -156,180 +140,97 @@ func UpdateModelService(ctx context.Context, c dynamic.Interface, input generate
-	if input.ModelType == "" || input.ModelType == modelTypeAll || input.ModelType == modelTypeLLM {
-		for idx := range llmList.Nodes {
-			v := llmList.Nodes[idx].(*generated.Llm)
-			klog.V(5).Infof("add llm modelservice: llm %s, type: %s, provider: %s. filter apiType: %s, filter providers: %s",
-				v.Name, *v.Type, *v.Provider, input.APIType, input.ProviderType)
-			if input.APIType != nil && *v.Type != *input.APIType {
-				continue
-			}
-			if input.ProviderType != nil && *v.Provider != *input.ProviderType {
-				continue
-			}
-
-			ms := &generated.ModelService{
-				Name:              v.Name,
-				Namespace:         input.Namespace,
-				Creator:           v.Creator,
-				Description:       v.Description,
-				Types:             new(string),
-				CreationTimestamp: v.CreationTimestamp,
-				UpdateTimestamp:   v.UpdateTimestamp,
-				APIType:           v.Type,
-				LlmResource:       v,
-				EmbedderResource:  new(generated.Embedder),
-				Resource:          new(generated.Resources),
-			}
-			*ms.Types = modelTypeLLM
-			intersec[v.Name] = ms
-
-			modelServiceList = append(modelServiceList, ms)
-			klog.V(5).Infof("add llm modelservice: append only llm modelService to list: %s", debugModelService(ms))
-			if r, ok := workerResource[v.Name]; ok {
-				klog.V(5).Infof(" set llm modelservice: %s resource: set modelservice resource: %+v", v.Name, r)
-				*ms.Resource = r.Resources
-				ms.ID = r.ID
-			}
-		}
-	} else {
-		for idx := range embedderList.Nodes {
-			v := embedderList.Nodes[idx].(*generated.Embedder)
-			klog.V(5).Infof("add embedder modelservice: embedder %s, type: %s, provider: %s. filter apiType: %s, filter providers: %s",
-				v.Name, *v.Type, *v.Provider, input.APIType, input.ProviderType)
-
-			if input.APIType != nil && *v.Type != *input.APIType {
-				continue
-			}
-			if input.ProviderType != nil && *v.Provider != *input.ProviderType {
-				continue
-			}
-
-			ms := &generated.ModelService{
-				Name:              v.Name,
-				Namespace:         input.Namespace,
-				Creator:           v.Creator,
-				Description:       v.Description,
-				Types:             new(string),
-				CreationTimestamp: v.CreationTimestamp,
-				UpdateTimestamp:   v.UpdateTimestamp,
-				APIType:           v.Type,
-				EmbedderResource:  v,
-				LlmResource:       new(generated.Llm),
-				Resource:          new(generated.Resources),
-			}
-			*ms.Types = modelTypeEmbedding
-
-			intersec[v.Name] = ms
-			modelServiceList = append(modelServiceList, ms)
-			klog.V(5).Infof("add embedder modelservice: append only embedder modelService to list: %s", debugModelService(ms))
-
-			if r, ok := workerResource[v.Name]; ok {
-				klog.V(5).Infof("set embedder modelservice: %s resource: %+v", v.Name, r)
-				*ms.Resource = r.Resources
-				ms.ID = r.ID
-			}
+	// sort by creation timestamp
+	llmList.Nodes = append(llmList.Nodes, embedderList.Nodes...)
+	sort.Slice(llmList.Nodes, func(i, j int) bool {
```

Comment:
@0xff-dev  We can use the `CreationTimestamp` which has been pased into `ModelService `  for this `sort`.

Reply:
@0xff-dev  add a `newNodeList` for sort